backend/src/main/java/ai/verta/modeldb/experimentRun/subtypes/KeyValueHandler.java

```diff
@@ -123,10 +134,16 @@ public KeyValueHandler(Executor executor, FutureJdbi jdbi, String fieldType, Str
     // Validate input
     return InternalFuture.runAsync(
             () -> {
+              Set<String> keySet = new HashSet<>();
               for (final var kv : kvs) {
                 if (kv.getKey().isEmpty()) {
                   throw new InvalidArgumentException("Empty key");
                 }
+                if (keySet.contains(kv.getKey())) {
+                  throw new InvalidArgumentException(
+                      "Multiple Key " + kv.getKey() + " found in " + getTableName());
```

Comment:
```suggestion
                      "Multiple key " + kv.getKey() + " found in request");
```
The keys are coming directly from the request, not the table.

Reply:
Fixed